src/OpenLoco/src/Drawing/SoftwareDrawingEngine.cpp

```diff
@@ -309,33 +378,30 @@ namespace OpenLoco::Drawing
             SDL_UnlockSurface(_screenSurface);
         }
 
-        auto scaleFactor = Config::get().scaleFactor;
-        if (scaleFactor == 1 || scaleFactor <= 0)
-        {
-            if (SDL_BlitSurface(_screenSurface, nullptr, SDL_GetWindowSurface(_window), nullptr))
-            {
-                Logging::error("SDL_BlitSurface {}", SDL_GetError());
-                exit(1);
-            }
-        }
-        else
+        // Convert colors via palette mapping onto the RGBA surface.
+        if (SDL_BlitSurface(_screenSurface, nullptr, _screenRGBASurface, nullptr))
         {
-            // first blit to rgba surface to change the pixel format
-            if (SDL_BlitSurface(_screenSurface, nullptr, _screenRGBASurface, nullptr))
-            {
-                Logging::error("SDL_BlitSurface {}", SDL_GetError());
-                exit(1);
-            }
-            // then scale to window size. Without changing to RGBA first, SDL complains
-            // about blit configurations being incompatible.
-            if (SDL_BlitScaled(_screenRGBASurface, nullptr, SDL_GetWindowSurface(_window), nullptr))
-            {
-                Logging::error("SDL_BlitScaled {}", SDL_GetError());
-                exit(1);
-            }
+            Logging::error("SDL_BlitSurface {}", SDL_GetError());
+            exit(1);
         }
 
-        SDL_UpdateWindowSurface(_window);
+        // Stream the RGBA pixels into unscaled screen texture.
+        void* pixels;
+        int pitch;
+        SDL_LockTexture(_screenTexture, NULL, &pixels, &pitch);
+        SDL_ConvertPixels(_screenRGBASurface->w, _screenRGBASurface->h, _screenRGBASurface->format->format, _screenRGBASurface->pixels, _screenRGBASurface->pitch, _screenTextureFormat->format, pixels, pitch);
+        SDL_UnlockTexture(_screenTexture);
+
+        // Copy screen texture to the scaled texture.
```

Comment:
Should make this a conditional again for when there is no scaling otherwise unnecessary copy.